core/templates/services/playthrough.service.ts

```diff
@@ -274,24 +299,39 @@ export class PlaythroughService {
    * If none of the issue types have been discovered, returns null instead.
    */
   private createNewPlaythrough(): Playthrough | null {
-    if (this.misTracker && this.misTracker.foundAnIssue()) {
-      return this.playthroughObjectFactory
-        .createNewMultipleIncorrectSubmissionsPlaythrough(
-          this.explorationId, this.explorationVersion,
-          this.misTracker.generateIssueCustomizationArgs(),
-          this.recordedLearnerActions);
-    } else if (this.cstTracker && this.cstTracker.foundAnIssue()) {
-      return this.playthroughObjectFactory
-        .createNewCyclicStateTransitionsPlaythrough(
-          this.explorationId, this.explorationVersion,
-          this.cstTracker.generateIssueCustomizationArgs(),
-          this.recordedLearnerActions);
-    } else if (this.eqTracker && this.eqTracker.foundAnIssue()) {
-      return this.playthroughObjectFactory
-        .createNewEarlyQuitPlaythrough(
-          this.explorationId, this.explorationVersion,
-          this.eqTracker.generateIssueCustomizationArgs(),
-          this.recordedLearnerActions);
+    // TODO(#15212): Remove the below check once codebase is strictly typed.
+    if (
+      this.explorationId &&
+      this.explorationVersion &&
+      this.recordedLearnerActions
+    ) {
+      if (this.misTracker && this.misTracker.foundAnIssue()) {
+        return this.playthroughObjectFactory
+          .createNewMultipleIncorrectSubmissionsPlaythrough(
+            this.explorationId, this.explorationVersion,
+            this.misTracker.generateIssueCustomizationArgs(),
+            this.recordedLearnerActions);
+      } else if (this.cstTracker && this.cstTracker.foundAnIssue()) {
+        const cstTrackerIssueCustomizationArgs = (
```

Comment:
Just to check -- how is it possible for this call to result in undefined? (if you look at the impl)

Ditto below.

Reply:
Can you explain more precisely the flow through which this would happen?

(The argument "just for safe side" could be a slippery slope; it would imply that we should be adding `!== undefined` to pretty much everything. So I want to understand how it is possible that AngularJS could actually send undefined here -- it would be better to make the necessary changes and then "prove" that such a value is impossible.)

Reply:
After investigation i found, it is not possible this call result into undefined as all other services uses this service has been already migrated and files uses those services not send undefined to it.

Hence REMOVED.